scipy/signal/_spectral_py.py

```diff
@@ -674,21 +687,60 @@ def spectrogram(x, fs=1.0, window=('tukey', .25), nperseg=None, noverlap=None,
 
     See Also
     --------
+    stft: Compute the Short-time Fourier Transform (STFT)
     periodogram: Simple, optionally modified periodogram
     lombscargle: Lomb-Scargle periodogram for unevenly sampled data
     welch: Power spectral density by Welch's method.
     csd: Cross spectral density by Welch's method.
 
     Notes
     -----
+    The spectrogram can be interpreted as a wrapper of the short-time
+    Fourier Transform (`stft`). The `stft` is more flexible in specifying the
+    boundary behavior and always retuns complex values (``mode=complex``).
+    Common uses of spectrograms are investigations of the evolution of a
+    magnitude spectrum  (``mode='magnitude', scaling='magnitude'``) or of a
+    power spectral density (``mode='psd', scaling='density'``) over time.
+
     An appropriate amount of overlap will depend on the choice of window
-    and on your requirements. In contrast to welch's method, where the
+    and on your requirements. In contrast to Welch's method, where the
     entire data stream is averaged over, one may wish to use a smaller
     overlap (or perhaps none at all) when computing a spectrogram, to
     maintain some statistical independence between individual segments.
     It is for this reason that the default window is a Tukey window with
     1/8th of a window's length overlap at each end.
 
+    Not all combinations of the parameters `mode`, `scaling` and
+    `return_onesided` allow a straight-forward interpretation. As an example
+    a 10-second single cosine test signal with amplitude 10 V is investigated:
+    Its power is 50 V\ :sup:`2` (``sum(x**2)/len(x)``). The following table
+    shows how the maximum and the power of the spectrogram depend on the input
+    parameters:
+
+    .. code-block:: text
+
+        mode       scaling    return_onesided  max(S)    Power
+        =========  =========  ===============  ========  =====
+        psd        density         True        113.4      50
+        psd        density         False       56.7       50
+        psd        spectrum        True        50.0       22
+        psd        spectrum        False       25.0       22
+        complex    density         True        7.5+0.0j   94.7
+        complex    density         False       7.5+0.0j  378.8
+        complex    spectrum        True        5.0+0.0j   41.7
+        complex    spectrum        False       5.0+0.0j  167
+        magnitude  density         True        7.5        94.7
+        magnitude  density         False       7.5       378.8
```

Comment:
Why not a proper table?

https://docutils.sourceforge.io/docs/user/rst/quickref.html#tables

Reply:
It's a matter of taste I guess. The text version only needs half the vertical space:

![image](https://user-images.githubusercontent.com/12721170/165146995-6ef88cec-84a9-4a06-85c3-5acf12f95566.png)
versus
![image](https://user-images.githubusercontent.com/12721170/165146874-7c82b7f2-5d30-4667-a451-88cc5ba51ddb.png)

In this case, I think a more compact table is better, since the information it contains is not that central to the topic. I don't have very strong opinions on this, so I'm fine with both versions. What do you think @larsoner?

Reply:
If you try a
```
.. table::
   :class: table-sm
```
does it look better from a spacing perspective? It just seems like a shame to use a non-table directive for a table...

Reply:
That's much better. Fixed it.Thanks.